### PR TITLE
implement additional tracker methods

### PR DESCRIPTION
**Pull request by pheel**
**Proposed changes**:
The Rasa tracker class implements the convenience methods `last_executed_action_has`, `get_last_event_for` and `applied_events`. This PR simply copies over this functionality from Rasa.

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa-sdk#code-style) for instructions)

### CODE REVIEW

File: rasa_sdk/interfaces.py [hunk 1]
```diff
@@ -147,6 +149,54 @@ def copy(self) -> "Tracker":
             self.latest_action_name,
         )
 
+    def last_executed_action_has(self, name, skip=0):
```
Comment (erohmensing) [inline rasa_sdk/interfaces.py]: Docstring and types please

File: rasa_sdk/interfaces.py [hunk 2]
```diff
@@ -147,6 +149,54 @@ def copy(self) -> "Tracker":
...
+        last = self.get_last_event_for(
+            "action", exclude=[ACTION_LISTEN_NAME], skip=skip
+        )
+        return last is not None and last["name"] == name
+
+    def get_last_event_for(self, event_type, exclude=[], skip=0):
```
Comment (erohmensing) [inline rasa_sdk/interfaces.py]: docstring and types

File: rasa_sdk/interfaces.py [hunk 3]
```diff
@@ -147,6 +149,54 @@ def copy(self) -> "Tracker":
...
+        filtered = filter(filter_function, reversed(self.applied_events()))
+        for _ in range(skip):
+            next(filtered, None)
+
+        return next(filtered, None)
+
+    def applied_events(self) -> List[Dict]:
+        """Returns all actions that should be applied - w/o reverted events."""
+
+        def undo_till_previous(event_type, done_events):
```
Comment (erohmensing) [inline rasa_sdk/interfaces.py]: types (especially important as the event_type is not the same as in the rasa code)

File: rasa_sdk/interfaces.py [hunk 4]
```diff
@@ -147,6 +149,54 @@ def copy(self) -> "Tracker":
...
+                occurrence `event_type` is found which is also removed."""
+            # list gets modified - hence we need to copy events!
+            for e in reversed(done_events[:]):
+                del done_events[-1]
+                if e["event"] == event_type:
+                    break
+
+        applied_events = []
+        for event in self.events:
+            if event.get("name", None) == "restart":
```
Comment (erohmensing) [inline rasa_sdk/interfaces.py]:
```suggestion
            if event.get("name") == "restart":
```
default for `.get()` is already `None`, thus we can remove these :)

File: CHANGELOG.rst [hunk 5]
```diff
@@ -12,6 +12,11 @@ This project adheres to `Semantic Versioning`_ starting with version 0.11.0.
 
 .. note:: This version is not yet released and is under active development.
 
+Added
+-----
+- Copied over instance methods `last_executed_action_has`, `get_last_event_for` and
```
Comment (erohmensing) [inline CHANGELOG.rst]:
```suggestion
- Copied over instance methods ``last_executed_action_has()``, ``get_last_event_for()`` and
```

File: CHANGELOG.rst [hunk 6]
```diff
@@ -12,6 +12,11 @@ This project adheres to `Semantic Versioning`_ starting with version 0.11.0.
...
+  `applied_events` from the Rasa Tracker class to the SDK Tracker class.
```
Comment (erohmensing) [inline CHANGELOG.rst]:
```suggestion
  ``applied_events`` from the Rasa ``DialogueStateTracker`` class to the SDK ``Tracker`` class.
```

**Comment by erohmensing** [comment]
@pheel are you still interested in making this PR happen? I think it's definitely a valuable addition.

**Comment by pheel** [comment]
@erohmensing Thanks for the reminder! I'll get back on it this week!

**Comment by erohmensing** [comment]
Awesome @pheel! Let me know if you have any questions 🙂

**Comment by pheel** [comment]
@erohmensing I've just updated the PR!

File: CHANGELOG.rst [hunk 7]
```diff
@@ -12,6 +12,15 @@ This project adheres to `Semantic Versioning`_ starting with version 0.11.0.
...
 
+[PR] - 2020-02-23
+^^^^^^^^^^^^^^^^^^^^
+
+Added
+-----
+- Copied over instance methods ``last_executed_action_has()``, ``get_last_event_for()`` and
+  ``applied_events`` from the Rasa ``DialogueStateTracker`` class to the SDK ``Tracker`` class.
+
+
```
Comment (erohmensing) [inline CHANGELOG.rst]: Read the note above, this file should not be edited directly. Please revert this change and add your entry based on instructions here: https://github.com/RasaHQ/rasa-sdk/tree/master/changelog

**Comment by erohmensing** [comment]
Went ahead and fixed the changelog so that this can make the 1.8 release train. Thanks for your contribution!
